src/azure-cli/azure/cli/command_modules/appservice/_validators.py

```diff
@@ -71,3 +71,26 @@ def validate_asp_create(cmd, namespace):
         validation = client.validate(resource_group_name, validation_payload)
         if validation.status.lower() == "failure" and validation.error.code != 'ServerFarmAlreadyExists':
             raise CLIError(validation.error.message)
+
+
+def validate_add_vnet(cmd, namespace):
+    if isinstance(namespace.name, str) and isinstance(namespace.resource_group, str):
+        resource_group_name = namespace.resource_group
+        from azure.cli.command_modules.network._client_factory import network_client_factory
+        vnet_client = network_client_factory(cmd.cli_ctx)
+        list_all_vnets = vnet_client.virtual_networks.list_all()
+        vnet = namespace.vnet
+        name = namespace.name
+        slot = namespace.slot
+
+        vnet_loc = ''
+        for v in list_all_vnets:
+            if v.name == vnet:
+                vnet_loc = v.location
+
+        from ._appservice_utils import _generic_site_operation
+        webapp = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot)
+        webapp_loc = webapp.location
+
+        if vnet_loc != webapp_loc:
+            raise CLIError("cannot add a regional vnet integration in a different region")
```

Comment:
The app and the vnet resources are in different locations. Cannot integrate a regional VNET  to an app in a different region.